core/src/main/scala/kafka/server/KafkaRequestHandler.scala

```diff
@@ -40,6 +40,8 @@ class KafkaRequestHandler(id: Int,
                           apis: KafkaApis,
                           time: Time) extends Runnable with Logging {
   this.logIdent = "[Kafka Request Handler " + id + " on Broker " + brokerId + "], "
+  // the requestsHandled counter is defined only for testing
+  var requestsHandled: Long = 0L
```

Comment:
We generally don't want to add state/variable specifically for unit test, particularly if there is no existing code which does this.

Usually we can do unit test using mockit to verify that something is triggered. Can you see if this is doable? Also, maybe we can just read the newly defined metrics value to verify that the newly defined processor/acceptor has processed some requests.